### PR TITLE
Add zero check to checked_u8_power() and checked_u8_mul() in curve/stable.rs

### PR DESCRIPTION
**Pull request by vni**
Added zero check info few functions in stable.rs.

Actually, I was just lurking the source code. And do not know if it's truly needed.

### CODE REVIEW

File: token-swap/program/src/curve/stable.rs [hunk 1]
```diff
@@ -36,15 +36,19 @@ fn compute_a(amp: u64) -> Option<u64> {
 
 /// Returns self to the power of b
 fn checked_u8_power(a: &U256, b: u8) -> Option<U256> {
-    let mut result = *a;
-    for _ in 1..b {
+    let mut result = U256::from(1);
+    for _ in 1..=b {
```
Comment (joncinque) [inline token-swap/program/src/curve/stable.rs:40]: if `b = 0`, shouldn't it be better to do ?
```suggestion
    for _ in 0..b {
```

**Review by joncinque — COMMENTED** [review]
Thanks for the contribution! Any chance we could add a couple of tests for this?

File: token-swap/program/src/curve/stable.rs (hunk 1 above)
Comment (vni) [reply token-swap/program/src/curve/stable.rs:40]: yes, thanks, it looks nicer

**Comment by joncinque** [comment]
Would you mind adding a test for each of these cases that you're fixing?

**Comment by vni** [comment]
Hi!
Thanks about the test suggestions. I will check it in the nearest few days, but don't want to make any promises right now. : )

**Comment by joncinque** [comment]
Closing this since the stable curve was removed

**Pull request closed without merging**